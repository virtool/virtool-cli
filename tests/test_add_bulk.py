import pytest
import shutil
import subprocess

from syrupy import SnapshotAssertion
from syrupy.filters import props


@pytest.fixture
def precached_repo(empty_repo, test_files_path):
    shutil.copytree(
        (test_files_path / "cache_test"),
        (empty_repo.path / ".cache/ncbi"),
        dirs_exist_ok=True,
    )

    yield empty_repo

    shutil.rmtree(empty_repo.path)


@pytest.fixture
def ref_table_of_contents() -> dict[int, list[str]]:
    return {
        96892: ["NC_005954"],
        683179: ["NC_013464"],
        438782: [
            "NC_010314",
            "NC_010318",
            "NC_010316",
            "NC_010319",
            "NC_010315",
            "NC_010317",
            "EF546804.1",
            "EF546802.1",
            "EF546803.1",
            "EF546806.1",
            "EF546807.1",
            "EF546805.1",
        ],
        1169032: ["MH200607.1", "NC_003355", "KJ207375.1", "MK431779.1", "AB017504.1"],
        270478: ["NC_011560"],
        1468454: ["NC_023641"],
        1278205: ["NC_020160"],
        1505530: ["NC_024301"],
        48201: ["NC_043133.1"],
        1468172: ["NC_023892"],
        946046: ["NC_014791"],
        1441799: ["NC_023881"],
        132477: ["NC_013006"],
        1414644: ["NC_022745"],
        56879: ["NC_001793"],
        223262: ["NC_004630", "NC_004625"],
        2060511: ["NC_036587"],
        1324128: ["NC_021786"],
        345184: ["DQ178614", "DQ178613", "DQ178610", "DQ178611"],
        430059: ["NC_043170.1"],
        2170135: ["NC_030236.1"],
        1561150: ["NC_026472"],
        1198450: ["NC_038796.1", "NC_038797"],
        662596: ["NC_010239"],
    }


@pytest.mark.ncbi()
def test_add_ref_mini(
    ref_table_of_contents, precached_repo, snapshot: SnapshotAssertion
):
    for taxid in ref_table_of_contents:
        subprocess.run(
            [
                "virtool",
                "ref",
                "otu",
                "create",
                "--path",
                str(precached_repo.path),
                str(taxid),
            ],
            check=False,
        )

        subprocess.run(
            [
                "virtool",
                "ref",
                "sequences",
                "add",
                "--path",
                str(precached_repo.path),
                "--taxid",
                str(taxid),
            ]
            + ref_table_of_contents[taxid],
            check=False,
        )

    for taxid in ref_table_of_contents:
        otu = precached_repo.get_otu_by_taxid(taxid)
        assert otu.dict() == snapshot(exclude=props("id", "isolates"))
        assert otu.accessions == snapshot

        assert otu.accessions == set(
            accession.split(".")[0] for accession in ref_table_of_contents[taxid]
        )
