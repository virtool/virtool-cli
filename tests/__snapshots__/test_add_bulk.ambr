# serializer version: 1
# name: test_add_ref_mini
  dict({
    'acronym': '',
    'excluded_accessions': list([
    ]),
    'legacy_id': None,
    'molecule': None,
    'name': 'Nanovirus-like particle',
    'schema': list([
    ]),
    'taxid': 96892,
  })
# ---
# name: test_add_ref_mini.1
  set({
    'NC_005954',
  })
# ---
# name: test_add_ref_mini.10
  dict({
    'acronym': '',
    'excluded_accessions': list([
    ]),
    'legacy_id': None,
    'molecule': None,
    'name': 'Lagenaria siceraria endornavirus-California',
    'schema': list([
    ]),
    'taxid': 1468454,
  })
# ---
# name: test_add_ref_mini.11
  set({
    'NC_023641',
  })
# ---
# name: test_add_ref_mini.12
  dict({
    'acronym': '',
    'excluded_accessions': list([
    ]),
    'legacy_id': None,
    'molecule': None,
    'name': 'Dahlia latent viroid',
    'schema': list([
    ]),
    'taxid': 1278205,
  })
# ---
# name: test_add_ref_mini.13
  set({
    'NC_020160',
  })
# ---
# name: test_add_ref_mini.14
  dict({
    'acronym': '',
    'excluded_accessions': list([
    ]),
    'legacy_id': None,
    'molecule': None,
    'name': 'Pagoda yellow mosaic associated virus',
    'schema': list([
    ]),
    'taxid': 1505530,
  })
# ---
# name: test_add_ref_mini.15
  set({
    'NC_024301',
  })
# ---
# name: test_add_ref_mini.16
  dict({
    'acronym': '',
    'excluded_accessions': list([
    ]),
    'legacy_id': None,
    'molecule': None,
    'name': 'Maclura mosaic virus',
    'schema': list([
    ]),
    'taxid': 48201,
  })
# ---
# name: test_add_ref_mini.17
  set({
    'NC_043133',
  })
# ---
# name: test_add_ref_mini.18
  dict({
    'acronym': '',
    'excluded_accessions': list([
    ]),
    'legacy_id': None,
    'molecule': None,
    'name': 'Gaillardia latent virus',
    'schema': list([
    ]),
    'taxid': 1468172,
  })
# ---
# name: test_add_ref_mini.19
  set({
    'NC_023892',
  })
# ---
# name: test_add_ref_mini.2
  dict({
    'acronym': '',
    'excluded_accessions': list([
    ]),
    'legacy_id': None,
    'molecule': None,
    'name': 'Saccharum streak virus',
    'schema': list([
    ]),
    'taxid': 683179,
  })
# ---
# name: test_add_ref_mini.20
  dict({
    'acronym': '',
    'excluded_accessions': list([
    ]),
    'legacy_id': None,
    'molecule': None,
    'name': 'Ugandan cassava brown streak virus',
    'schema': list([
    ]),
    'taxid': 946046,
  })
# ---
# name: test_add_ref_mini.21
  set({
    'NC_014791',
  })
# ---
# name: test_add_ref_mini.22
  dict({
    'acronym': '',
    'excluded_accessions': list([
    ]),
    'legacy_id': None,
    'molecule': None,
    'name': 'Faba bean necrotic stunt alphasatellite 1',
    'schema': list([
    ]),
    'taxid': 1441799,
  })
# ---
# name: test_add_ref_mini.23
  set({
    'NC_023881',
  })
# ---
# name: test_add_ref_mini.24
  dict({
    'acronym': '',
    'excluded_accessions': list([
    ]),
    'legacy_id': None,
    'molecule': None,
    'name': 'Kalanchoe latent virus',
    'schema': list([
    ]),
    'taxid': 132477,
  })
# ---
# name: test_add_ref_mini.25
  set({
    'NC_013006',
  })
# ---
# name: test_add_ref_mini.26
  dict({
    'acronym': '',
    'excluded_accessions': list([
    ]),
    'legacy_id': None,
    'molecule': None,
    'name': 'Tall oatgrass mosaic virus',
    'schema': list([
    ]),
    'taxid': 1414644,
  })
# ---
# name: test_add_ref_mini.27
  set({
    'NC_022745',
  })
# ---
# name: test_add_ref_mini.28
  dict({
    'acronym': '',
    'excluded_accessions': list([
    ]),
    'legacy_id': None,
    'molecule': None,
    'name': 'Oat blue dwarf virus',
    'schema': list([
    ]),
    'taxid': 56879,
  })
# ---
# name: test_add_ref_mini.29
  set({
    'NC_001793',
  })
# ---
# name: test_add_ref_mini.3
  set({
    'NC_013464',
  })
# ---
# name: test_add_ref_mini.30
  dict({
    'acronym': '',
    'excluded_accessions': list([
    ]),
    'legacy_id': None,
    'molecule': None,
    'name': 'East African cassava mosaic Cameroon virus',
    'schema': list([
    ]),
    'taxid': 223262,
  })
# ---
# name: test_add_ref_mini.31
  set({
    'NC_004625',
    'NC_004630',
  })
# ---
# name: test_add_ref_mini.32
  dict({
    'acronym': '',
    'excluded_accessions': list([
    ]),
    'legacy_id': None,
    'molecule': None,
    'name': 'Babaco mosaic virus',
    'schema': list([
    ]),
    'taxid': 2060511,
  })
# ---
# name: test_add_ref_mini.33
  set({
    'NC_036587',
  })
# ---
# name: test_add_ref_mini.34
  dict({
    'acronym': '',
    'excluded_accessions': list([
    ]),
    'legacy_id': None,
    'molecule': None,
    'name': 'Habenaria mosaic virus',
    'schema': list([
    ]),
    'taxid': 1324128,
  })
# ---
# name: test_add_ref_mini.35
  set({
    'NC_021786',
  })
# ---
# name: test_add_ref_mini.36
  dict({
    'acronym': '',
    'excluded_accessions': list([
    ]),
    'legacy_id': None,
    'molecule': None,
    'name': 'Cabbage leaf curl Jamaica virus',
    'schema': list([
    ]),
    'taxid': 345184,
  })
# ---
# name: test_add_ref_mini.37
  set({
    'DQ178610',
    'DQ178611',
    'DQ178613',
    'DQ178614',
  })
# ---
# name: test_add_ref_mini.38
  dict({
    'acronym': '',
    'excluded_accessions': list([
    ]),
    'legacy_id': None,
    'molecule': None,
    'name': 'Vallota mosaic virus',
    'schema': list([
    ]),
    'taxid': 430059,
  })
# ---
# name: test_add_ref_mini.39
  set({
    'NC_043170',
  })
# ---
# name: test_add_ref_mini.4
  dict({
    'acronym': '',
    'excluded_accessions': list([
    ]),
    'legacy_id': None,
    'molecule': None,
    'name': 'Abaca bunchy top virus',
    'schema': list([
    ]),
    'taxid': 438782,
  })
# ---
# name: test_add_ref_mini.40
  dict({
    'acronym': '',
    'excluded_accessions': list([
    ]),
    'legacy_id': None,
    'molecule': None,
    'name': 'Impatiens flower break virus',
    'schema': list([
    ]),
    'taxid': 2170135,
  })
# ---
# name: test_add_ref_mini.41
  set({
    'NC_030236',
  })
# ---
# name: test_add_ref_mini.42
  dict({
    'acronym': '',
    'excluded_accessions': list([
    ]),
    'legacy_id': None,
    'molecule': None,
    'name': 'Yacon necrotic mottle virus',
    'schema': list([
    ]),
    'taxid': 1561150,
  })
# ---
# name: test_add_ref_mini.43
  set({
    'NC_026472',
  })
# ---
# name: test_add_ref_mini.44
  dict({
    'acronym': '',
    'excluded_accessions': list([
    ]),
    'legacy_id': None,
    'molecule': None,
    'name': 'Jacquemontia mosaic Yucatan virus',
    'schema': list([
    ]),
    'taxid': 1198450,
  })
# ---
# name: test_add_ref_mini.45
  set({
    'NC_038796',
    'NC_038797',
  })
# ---
# name: test_add_ref_mini.46
  dict({
    'acronym': '',
    'excluded_accessions': list([
    ]),
    'legacy_id': None,
    'molecule': None,
    'name': 'Radish leaf curl betasatellite',
    'schema': list([
    ]),
    'taxid': 662596,
  })
# ---
# name: test_add_ref_mini.47
  set({
    'NC_010239',
  })
# ---
# name: test_add_ref_mini.5
  set({
    'EF546802',
    'EF546803',
    'EF546804',
    'EF546805',
    'EF546806',
    'EF546807',
    'NC_010314',
    'NC_010315',
    'NC_010316',
    'NC_010317',
    'NC_010318',
    'NC_010319',
  })
# ---
# name: test_add_ref_mini.6
  dict({
    'acronym': '',
    'excluded_accessions': list([
    ]),
    'legacy_id': None,
    'molecule': None,
    'name': 'Wasabi mottle virus',
    'schema': list([
    ]),
    'taxid': 1169032,
  })
# ---
# name: test_add_ref_mini.7
  set({
    'AB017504',
    'KJ207375',
    'MH200607',
    'MK431779',
    'NC_003355',
  })
# ---
# name: test_add_ref_mini.8
  dict({
    'acronym': '',
    'excluded_accessions': list([
    ]),
    'legacy_id': None,
    'molecule': None,
    'name': 'Zantedeschia mild mosaic virus',
    'schema': list([
    ]),
    'taxid': 270478,
  })
# ---
# name: test_add_ref_mini.9
  set({
    'NC_011560',
  })
# ---
